presto-main/src/main/java/com/facebook/presto/execution/StageExecutionStateMachine.java

```diff
@@ -369,16 +372,16 @@ public void recordGetSplitTime(long startNanos)
         runtimeStats.addMetricValue(GET_SPLITS_TIME_NANOS, NANO, elapsedNanos);
     }
 
-    public void recordSchedulerRunningTime(long cpuTimeNanos, long wallTimeNanos)
+    public void recordSchedulerRunningTime(long cpuTimeNanos, long wallTimeNanos, boolean isScanStage)
```

Comment:
nit: instead of passing a flag should we add a new method for recording the new metric and call that directly in SqlStageExecution? I'm not sure if we're saving much by passing this off until later.

Reply:
We dont have access to the runtime stats here, hence we are doing it in the StageExecutionStateMachine

Reply:
I meant have two methods in StageExecutionStateMachine - recordSchedulerRunningTime() and recordLeafStageSchedulerRunningTime(). And then SqlStageExecution would call the right one depending on if it's a leaf stage or not.  That way StageExecutionStateMachine doesn't need to know anything about the different stages and doesn't have any branching. It just updates the metric that it's asked to update.